### PR TITLE
std::net: Socket::new_raw now set to SO_NOSIGPIPE on freebsd.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: library/std/src/sys/pal/unix/net.rs [hunk 1]
```diff
@@ -74,7 +74,6 @@ impl Socket {
                 if #[cfg(any(
                     target_os = "android",
                     target_os = "dragonfly",
-                    target_os = "freebsd",
```
Comment (taiki-e) [inline library/std/src/sys/pal/unix/net.rs:77]: This makes an extra syscall to call set_cloexec necessary, right?

**Comment by taiki-e** [comment]
Btw, NetBSD and DragonFly BSD also support nosigpipe: https://github.com/bytecodealliance/rustix/pull/870

File: library/std/src/sys/pal/unix/net.rs (hunk 1 above)
Comment (devnexen) [reply library/std/src/sys/pal/unix/net.rs:77]: yes

**Comment by workingjubilee** [comment]
@devnexen Thanks for the PR! Pardon me, I seem to be a bit dense today. What does the part of the PR title that says "no set to" mean?

**Comment by devnexen** [comment]
Hehe just typo. meant "noW set to"

**Comment by workingjubilee** [comment]
And it seems this is already done, albeit with very slight API differences, by The Other BSD. Cool, safe to continue doing platform-specific stuff then.

@bors r+ rollup

**Comment by workingjubilee** [comment]
r? workingjubilee

**Pull request merged**